### PR TITLE
SFINT-4717 logCaseAttach and logCaseDetach added to insight analytics client

### PR DESCRIPTION
**Pull request by mmitiche**
[SFINT-4717](https://coveord.atlassian.net/browse/SFINT-4717)

## New methods added to the Insight Analytics client:

- ### logCaseAttach: this method will log the following click event:
AttachToCase

- ### logCaseDetach: this method will log the following custom event:
CaseDetach

### CODE REVIEW

File: src/searchPage/searchPageEvents.ts [hunk 1]
```diff
@@ -314,6 +322,7 @@ export const CustomEventsTypes: Partial<Record<SearchPageEvents | InsightEvents,
     [SearchPageEvents.clearRecentResults]: 'recentlyClickedDocuments',
     [SearchPageEvents.showLessFoldedResults]: 'folding',
     [InsightEvents.expandToFullUI]: 'interface',
+    [SearchPageEvents.caseDetach]: 'case',
```
Comment (nathanlb) [inline src/searchPage/searchPageEvents.ts:325]: We don't need to add `caseAttach` here?

**Review by nathanlb — APPROVED** [review]
Looks legit to me

File: src/searchPage/searchPageEvents.ts (hunk 1 above)
Comment (mmitiche) [reply src/searchPage/searchPageEvents.ts:325]: No need, this is the `CustomEventsTypes` object,  we only use it in custom events. Case detach is a custom event but the case attach event isn't.

Comment (mmitiche) [reply src/searchPage/searchPageEvents.ts:325]: nope didn't find any, we currently do NOT send any action type with click events.

File: src/insight/insightClient.ts [hunk 2]
```diff
@@ -258,6 +258,32 @@ export class CoveoInsightClient {
...
+        const metadata = {
+            documentTitle: info.documentTitle,
+            documentURL: info.documentUrl,
+            resultUriHash: info.documentUriHash,
+        };
+
+        return this.logClickEvent(
+            SearchPageEvents.caseAttach,
+            info,
+            identifier,
```
[the remark this answers is not in the document]
Comment (mmitiche) [reply src/insight/insightClient.ts:275]: It's the following:
Identifier

File: src/insight/insightClient.ts [hunk 3]
```diff
@@ -258,6 +258,32 @@ export class CoveoInsightClient {
...
 
+    public logCaseAttach(
+        info: PartialDocumentInformation,
+        identifier: DocumentIdentifier,
+        caseMetadata?: CaseMetadata
+    ) {
+        const metadata = {
+            documentTitle: info.documentTitle,
+            documentURL: info.documentUrl,
+            resultUriHash: info.documentUriHash,
```
[the remark this answers is not in the document]
Comment (erocheleau) [reply src/insight/insightClient.ts:269]: all good then it answers the question

**Review by erocheleau — APPROVED** [review]

**Pull request merged**